binary_search/binary_search.c

```diff
@@ -5,6 +5,40 @@
  * @arr_size Size of array
  * @search_element Element to be searched
  */
+ 
+/* function to perform recursive binary search
+ * input parameters:
+ * @ar - user-input array.
+ * @l - left index of array.
+ * @r - right index of array.
+ * @ele - element to be searched.
+ */
+int recursive_binary_search(const int *ar, int l, int r, int ele)
+{
```

Comment:
This opening bracket should start after the function name as `func_name(...) {` as per guidelines mentioned. Please change this for all conditional and looping statements too as `if (...) {` and `for (...) {`